pkg/printer/printer.go

```diff
@@ -2,31 +2,75 @@ package printer
 
 import (
 	"fmt"
+	"os"
 
 	"github.com/doitintl/kube-no-trouble/pkg/judge"
 )
 
-var printers = map[string]func() Printer{
+var printers = map[string]func(string) (Printer, error){
 	"json": newJSONPrinter,
 	"text": newTextPrinter,
 }
 
 type Printer interface {
 	Print([]judge.Result) error
+	Close() error
 }
 
-func NewPrinter(choice string) (Printer, error) {
+type commonPrinter struct {
+	outputFile *os.File
+}
+
+// newCommonPrinter creates new printer that prints to given output file
```

Comment:
these comments are not very useful

Reply:
Better suggestion? Explained above why I would like to have comments.